src/eventSubscription/commands/mock/createMockEventGenerator.ts

```diff
@@ -78,6 +102,12 @@ function getEventTypeFromTopic(topic: string): EventType {
         return EventType.Resources;
     } else if (/^\/subscriptions\/.*\/resourceGroups\/.*\/providers\/microsoft.storage\/storageaccounts\/[^\/]+$/i.test(topic)) {
         return EventType.Storage;
+    } else if (/^\/subscriptions\/.*\/resourceGroups\/.*\/providers\/microsoft.containerregistry\/registries\/[^\/]+$/i.test(topic)) {
+        return EventType.ContainerRegistry;
+    } else if (/^\/subscriptions\/.*\/resourceGroups\/.*\/providers\/microsoft.devices\/iothubs\/[^\/]+$/i.test(topic)) {
+        return EventType.Devices;
+    } else if (/^\/subscriptions\/.*\/resourceGroups\/.*\/providers\/microsoft.eventhub\/namespaces\/[^\/]+$/i.test(topic)) {
+        return EventType.EventHub;
```

Comment:
I'm seeing a lot of regex tests for the existence of 'subscriptions', 'resourceGroups', and 'providers'. 
It looks like repeated code prone to error, and performs repeated regex checks. 
Would it make more  enter these checks only if it looks like an Azure Resource URI (has the above tags)?